src/WRings.sol

```diff
@@ -0,0 +1,268 @@
+    function maxRedeem(address owner) external view returns(uint256) {
+        return _balanceOf[owner];
+    }
+
+    /// @notice used to simulate a deposit at the current block
+    /// @param assets amount of assets being deposited
+    /// @return shares amount of shares minted in exchange for assets
+    function previewDeposit(uint256 assets) external view returns(uint256) {
+        return _convertToShares(assets);
+    }
+
+    /// @notice used to simulate a deposit at the current block
+    /// @param shares desired amount of shares to mint
+    /// @return assets amount of asset token required for desired shares
+    function previewMint(uint256 shares) external view returns(uint256) {
+        return _convertToAssets(shares);
+    }
+
+    /// @notice used to simulate a withdrawal and the shares required
+    /// @param assets desired amount of asset token to receive
+    /// @return shares amount of shares required to redeem
+    function previewWithdraw(uint256 assets) external view returns(uint256) {
+        return _convertToShares(assets);
+    }
+
+    /**
+     * @dev Initiate a flash loan.
+     * @param receiver The receiver of the tokens in the loan, and the receiver of the callback.
+     * @param token The loan currency.
+     * @param amount The amount of tokens lent.
+     * @param data Arbitrary data structure, intended to contain user-defined parameters.
+     */
+    function flashLoan(
+        IERC3156FlashBorrower receiver,
+        address token,
+        uint256 amount,
+        bytes calldata data
+        ) external override returns (bool) {
+        require(token == address(iRings), "WRings: Unsupported token");
+        uint256 fee_ = _flashFee(amount);
+        require(
+            iRings.transfer(address(receiver), amount),
+            "WRings: Transfer failed"
+        );
+        require(
+            receiver.onFlashLoan(msg.sender, token, amount, fee_, data) == CALLBACK_SUCCESS,
+            "IERC3156: Callback failed"
+        );
+        uint256 _allowance = iRings.allowance(address(receiver), address(this));
+        require(
+            _allowance >= (amount + fee_),
+            "WRings: Repay not approved"
+        );
+        require(
+            iRings.transferFrom(address(receiver), address(this), amount + fee_),
+            "WRings: Repay failed"
+        );
+        return true;
+    }
+
+    function init(uint256 assets, address receiver) external returns(uint256) {
+        _mint(receiver, assets);
+        iRings.safeTransferFrom(msg.sender, address(this), assets);
+        return assets;
+    }
+
+     /// @notice User specifies amount of asset token to deposit in return for vault shares
+    /// @param assets Number of asset tokens to deposit
+    /// @param receiver Address that vault tokens will be minted to
+    /// @return shares Number of vault shares to return to the user
+    function deposit(uint256 assets, address receiver) external  returns(uint256) {
```

Comment:
```suggestion
    function deposit(uint256 assets, address receiver) external  returns(uint256 shares) {
```

I find the syntax cleaner if the return variables have names and we skip the `return` statement

Reply:
I think I'd adopted this pattern since writing the first version of the vault. But it didn't stand out to me on the second version enough to change it.